include/ccf/receipt.h

```diff
@@ -68,13 +71,16 @@ namespace ccf
       auto components = receipt.leaf_components.value();
       if (
         components.write_set_digest.has_value() &&
+        components.commit_evidence.has_value() &&
         components.claims_digest.has_value())
       {
         auto ws_dgst = crypto::Sha256Hash::from_hex_string(
           components.write_set_digest.value());
+        auto ce_dgst =
+          crypto::Sha256Hash::from_string(components.commit_evidence.value());
         auto cl_dgst =
           crypto::Sha256Hash::from_hex_string(components.claims_digest.value());
-        current = crypto::Sha256Hash(ws_dgst, cl_dgst);
+        current = crypto::Sha256Hash(ws_dgst, ce_dgst, cl_dgst);
       }
       else
       {
```

Comment:
Can't make a suggestion, but the error below should mention all 3 elements, or just "Cannot compute leaf unless all leaf elements are set".

Reply:
Done